metaci/testresults/tests/test_robot_importer.py

```diff
@@ -316,3 +318,119 @@ def test_import_perf_results():
         ("Set Time and Also Metric", 0.0),
     ]:
         assert durations[name] == value
+
+
+@responses.activate
+@pytest.mark.django_db
+def test_gus_bus_test_manager(mocker):
+    """Verifies that we import all tests in a suite"""
+    mocker.patch(
+        "metaci.testresults.robot_importer.settings",
+        METACI_RELEASE_WEBHOOK_URL="https://webhook",
+        METACI_RELEASE_WEBHOOK_ISSUER="MetaCI",
+        METACI_RELEASE_WEBHOOK_AUTH_KEY="test",
+    )
+    mocker.patch(
+        "metaci.build.flows.settings",
+        RESULT_EXPORT_ENABLED=True,
+    )
+    mocker.patch(
+        "metaci.testresults.tests.test_robot_importer.settings",
+        METACI_RELEASE_WEBHOOK_URL="https://webhook",
+    )
+    mocker.patch(
+        "metaci.release.utils.settings",
+        METACI_RELEASE_WEBHOOK_ISSUER="https://webhook",
+        METACI_RELEASE_WEBHOOK_AUTH_KEY="12345",
+    )
+
+    flow_task = FlowTaskFactory()
+    with temporary_dir() as output_dir:
+        copyfile(
+            TEST_ROBOT_OUTPUT_FILES / "robot_with_failures.xml",
+            Path(output_dir) / "output.xml",
+        )
+        robot_importer.import_robot_test_results(flow_task, output_dir)
+        responses.add(
+            "POST",
+            f"{settings.METACI_RELEASE_WEBHOOK_URL}/test-results/",
+            json={
+                "success": True,
+                "id": "123",
+            },
+        )
+        assert robot_importer.export_robot_test_results(FlowTaskFactory(), []) is None
+        assert len(responses.calls) == 1
+        assert responses.calls[0].request.url == "https://webhook/test-results/"
+
+
+@responses.activate
+@pytest.mark.django_db
+def test_gus_bus_test_manager_failure(mocker):
+    """Verifies that we import all tests in a suite"""
+    mocker.patch(
+        "metaci.testresults.robot_importer.settings",
+        METACI_RELEASE_WEBHOOK_URL="https://webhook",
+        METACI_RELEASE_WEBHOOK_ISSUER="MetaCI",
+        METACI_RELEASE_WEBHOOK_AUTH_KEY="test",
+    )
+    mocker.patch(
+        "metaci.build.flows.settings",
+        RESULT_EXPORT_ENABLED=True,
+    )
+    mocker.patch(
+        "metaci.testresults.tests.test_robot_importer.settings",
+        METACI_RELEASE_WEBHOOK_URL="https://webhook",
+    )
+    mocker.patch(
+        "metaci.release.utils.settings",
+        METACI_RELEASE_WEBHOOK_ISSUER="https://webhook",
+        METACI_RELEASE_WEBHOOK_AUTH_KEY="12345",
+    )
+
+    with pytest.raises(Exception):
```

Comment:
Please don't do this. It will pass if there's _any_ exception while running the test, even if it's not the one you expected. Put the pytest.raises context manager around one specific statement that you expect to fail, and if it's a generic exception, make sure you also use `match=` to confirm it's the message you expected.

Reply:
Thank you I've been looking for exactly what you instructed 🙏